### PR TITLE
Made assoc optional on add_news_items command

### PR DESCRIPTION
**Pull request by gsckoco**
Changes made based on #12979 
Added new method `bool News::CheckIfItemRequiresAssoc(News::ItemType type)`
Here's the new output in the debug console:
image

I have tested it and it works as expected with no errors in console, and as far as I can tell the new code follows the style guide.

### CODE REVIEW

**Comment by gsckoco** [comment]
Wait, ignore this request, after re-reading the original issue I realise you meant make the method have assoc

File: src/openrct2/management/NewsItem.cpp [hunk 1]
```diff
@@ -310,12 +310,34 @@ News::Item* News::AddItemToQueue(News::ItemType type, rct_string_id string_id, u
...
+ * @return A boolean if assoc is required.
+ */
+
+bool News::CheckIfItemRequiresAssoc(News::ItemType type)
+{
+    switch (type)
+    {
+        case News::ItemType::Null:
+            return false;
+        case News::ItemType::Award:
```
Comment (duncanspumpkin) [inline src/openrct2/management/NewsItem.cpp:324]: In general we would format this with all the false labels together. Rather than lots of return false.

**Review by duncanspumpkin — COMMENTED** [review]
This all looks sensible and correct. You will need to add a changelog entry but we are about to do a release today so wait till tomorrow for that as you will get a merge conflict. If this is your first contribution you should add yourself to contributors.md as a bug fixer.

**Review by tupaschoal — COMMENTED** [review]
Can you rebase and add the changelog entry, please?

This tutorial's got you covered: https://github.com/OpenRCT2/OpenRCT2/wiki/Rebase-and-Sync-fork-with-OpenRCT2

**Comment by gsckoco** [comment]
Yeah, I'll do this in the morning. Thanks for the link to the tutorial, I tried to add the changelog entry earlier as you had probably seen and failed 😂

Get Outlook for Android<https://aka.ms/ghei36>
________________________________

**Comment by gsckoco** [comment]
I have a feeling I have fucked this up quite bad, I am going to reopen this, hopefully with a better commit history.

**Comment by gsckoco** [comment]
New PR #13023

**Comment by duncanspumpkin** [comment]
Please don't close prs and reopen new ones. Almost everything does not require opening a new pr. We lose continuity with a new pr and i get spammed with extra notifications.

**Pull request closed without merging**